symfony/plugins/orangehrmTimePlugin/Dao/TimesheetDao.php

```diff
@@ -406,4 +408,95 @@ public function getDefaultTimesheet(
+    {
+        return $this->getTimesheetItemsPaginatorForEmployeeReport($filterParams)->count();
+    }
+
+    /**
+     * @param  EmployeeReportsSearchFilterParams  $filterParams
+     * @return Paginator
+     */
+    private function getTimesheetItemsPaginatorForEmployeeReport(
+        EmployeeReportsSearchFilterParams $filterParams
+    ): Paginator {
+        $qb = $this->getTimesheetItemsForEmployeeReportQueryBuilderWrapper($filterParams)->getQueryBuilder();
+        $qb->addSelect('COALESCE(SUM(timesheetItem.duration),0) AS totalDurationByGroup');
+        $qb->addGroupBy('timesheetItem.project');
+        $qb->addGroupBy('timesheetItem.projectActivity');
+        return $this->getPaginator($qb);
+    }
+
+    /**
+     * @param  EmployeeReportsSearchFilterParams  $filterParams
+     * @return QueryBuilderWrapper
+     */
+    private function getTimesheetItemsForEmployeeReportQueryBuilderWrapper(
+        EmployeeReportsSearchFilterParams $filterParams
+    ): QueryBuilderWrapper {
+        $q = $this->createQueryBuilder(TimesheetItem::class, 'timesheetItem');
+        $q->leftJoin('timesheetItem.timesheet', 'timesheet');
+        $q->leftJoin('timesheetItem.project', 'project');
+        $q->leftJoin('project.customer', 'customer');
+        $q->andWhere('timesheetItem.employee = :empNumber');
+        $q->setParameter('empNumber', $filterParams->getEmpNumber());
+        $this->setSortingAndPaginationParams($q, $filterParams);
+
+        if (!is_null($filterParams->getProjectId())) {
+            $q->andWhere('timesheetItem.project = :projectId');
+            $q->setParameter('projectId', $filterParams->getProjectId());
+        }
+
+        if (!is_null($filterParams->getActivityId())) {
+            $q->andWhere('timesheetItem.projectActivity = :activityId');
+            $q->setParameter('activityId', $filterParams->getActivityId());
+        }
+
+        //Timesheet items after fromDate (including fromDate)
+        if (!is_null($filterParams->getFromDate())) {
+            $q->andWhere($q->expr()->gte('timesheetItem.date', ':fromDate'));
+            $q->setParameter('fromDate', $filterParams->getFromDate());
+        }
+
+        //Timesheet items before toDate (including toDate)
+        if (!is_null($filterParams->getToDate())) {
+            $q->andWhere($q->expr()->lte('timesheetItem.date', ':toDate'));
+            $q->setParameter('toDate', $filterParams->getToDate());
+        }
+
+        if ($filterParams->getIncludeTimesheets(
+            ) === EmployeeReportsSearchFilterParams::INCLUDE_TIMESHEETS_APPROVED_ONLY) {
+            $q->andWhere('timesheet.state = :state');
+            $q->setParameter('state', EmployeeReportsSearchFilterParams::TIMESHEET_STATE[1]);
+        }
+
+        return $this->getQueryBuilderWrapper($q);
+    }
+
+    /**
+     * @param  EmployeeReportsSearchFilterParams  $filterParams
+     * @return string
+     */
```

Comment:
we can change the return type as int, then we no need to type caste inside the getMeta() method

Reply:
done